diff -purN otp_src_20.1.a/erts/emulator/drivers/common/efile_drv.c otp_src_20.1/erts/emulator/drivers/common/efile_drv.c
--- otp_src_20.1.a/erts/emulator/drivers/common/efile_drv.c	2017-09-25 22:10:38.000000000 +0300
+++ otp_src_20.1/erts/emulator/drivers/common/efile_drv.c	2017-12-05 18:28:42.186243705 +0300
@@ -4132,15 +4132,15 @@ file_outputv(ErlDrvData e, ErlIOVec *ev)
 	d->c.sendfile.port = desc->port;
 	d->c.sendfile.q_mtx = desc->q_mtx;
 
-    #if SIZEOF_OFF_T == 4
+    if (sizeof(off_t) == 4) {
 	if (offsetH != 0) {
 	    reply_posix_error(desc, EINVAL);
 	    goto done;
 	}
 	d->c.sendfile.offset = (off_t) offsetL;
-    #else
+    } else {
 	d->c.sendfile.offset = ((off_t) offsetH << 32) | offsetL;
-    #endif
+    }
 
 	d->c.sendfile.nbytes = nbytes;
 
diff -purN otp_src_20.1.a/erts/emulator/drivers/unix/unix_efile.c otp_src_20.1/erts/emulator/drivers/unix/unix_efile.c
--- otp_src_20.1.a/erts/emulator/drivers/unix/unix_efile.c	2017-09-25 22:10:38.000000000 +0300
+++ otp_src_20.1/erts/emulator/drivers/unix/unix_efile.c	2017-12-05 18:30:31.885311594 +0300
@@ -511,11 +511,11 @@ efile_fileinfo(Efile_error* errInfo, Efi
 	return 0;
     }
 
-#if SIZEOF_OFF_T == 4
-    pInfo->size_high = 0;
-#else
-    pInfo->size_high = (Uint32)(statbuf.st_size >> 32);
-#endif
+    if (sizeof(off_t) == 4) {
+        pInfo->size_high = 0;
+    } else {
+        pInfo->size_high = (Uint32)(statbuf.st_size >> 32);
+    }
     pInfo->size_low = (Uint32)statbuf.st_size;
 
 #ifdef NO_ACCESS
