diff -aur otp_src_17.3/erts/emulator/drivers/common/efile_drv.c otp_src_17.3-new/erts/emulator/drivers/common/efile_drv.c
--- otp_src_17.3/erts/emulator/drivers/common/efile_drv.c	2014-09-16 21:10:57.000000000 +0200
+++ otp_src_17.3-new/erts/emulator/drivers/common/efile_drv.c	2014-10-28 15:58:42.396163381 +0100
@@ -4164,15 +4164,15 @@
 	d->c.sendfile.port = desc->port;
 	d->c.sendfile.q_mtx = desc->q_mtx;
 
-    #if SIZEOF_OFF_T == 4
+    if (sizeof(off_t) == 4) {
 	if (offsetH != 0) {
 	    reply_posix_error(desc, EINVAL);
 	    goto done;
 	}
 	d->c.sendfile.offset = (off_t) offsetL;
-    #else
+    } else {
 	d->c.sendfile.offset = ((off_t) offsetH << 32) | offsetL;
-    #endif
+    }
 
 	d->c.sendfile.nbytes = nbytes;
 
diff -aur otp_src_17.3/erts/emulator/drivers/ose/ose_efile.c otp_src_17.3-new/erts/emulator/drivers/ose/ose_efile.c
--- otp_src_17.3/erts/emulator/drivers/ose/ose_efile.c	2014-09-16 21:10:57.000000000 +0200
+++ otp_src_17.3-new/erts/emulator/drivers/ose/ose_efile.c	2014-10-28 15:59:15.036162695 +0100
@@ -687,11 +687,11 @@
         return 0;
     }
 
-#if SIZEOF_OFF_T == 4
+if (sizeof(off_t) == 4) {
     pInfo->size_high = 0;
-#else
+} else {
     pInfo->size_high = (Uint32)(statbuf.st_size >> 32);
-#endif
+}
     pInfo->size_low = (Uint32)statbuf.st_size;
 
 #ifdef NO_ACCESS
diff -aur otp_src_17.3/erts/emulator/drivers/unix/unix_efile.c otp_src_17.3-new/erts/emulator/drivers/unix/unix_efile.c
--- otp_src_17.3/erts/emulator/drivers/unix/unix_efile.c	2014-09-16 21:10:57.000000000 +0200
+++ otp_src_17.3-new/erts/emulator/drivers/unix/unix_efile.c	2014-10-28 15:56:53.708165665 +0100
@@ -499,11 +499,11 @@
 	return 0;
     }
 
-#if SIZEOF_OFF_T == 4
+if (sizeof(off_t) == 4) {
     pInfo->size_high = 0;
-#else
+} else {
     pInfo->size_high = (Uint32)(statbuf.st_size >> 32);
-#endif
+}
     pInfo->size_low = (Uint32)statbuf.st_size;
 
 #ifdef NO_ACCESS
